tsdb/head.go

```diff
@@ -154,6 +154,9 @@ type HeadOptions struct {
 	EnableMemorySnapshotOnShutdown bool
 
 	IsolationDisabled bool
+
+	// Temporary flag which we use to select whether we want to use the new or the old chunk disk mapper.
```

Comment:
Can we clarify old/new terminology a bit?

```suggestion
	// Temporary flag which we use to select whether to use the new (asynchronous, used in upstream
	// Prometheus) or the old (legacy, synchronous) chunk disk mapper.
```

I understand that default value is chosen such that Mimir defaults to old (proven) mapper, but I find it little strange that mimir-prometheus defaults to "old" chunk mapper, which is no longer available in upstream Prometheus. If someone used mimir-prometheus instead of prometheus with no special settings, he would get different behaviour.

Reply:
> If someone used mimir-prometheus instead of prometheus with no special settings, he would get different behaviour.

That's true, but this is also the case already right now (before this PR). With this PR we basically just introduce to option for a user to switch to the same behavior as upstream if they want to. The plan is to roll this out in our production environments and then slowly enable the new chunk disk mapper everywhere, once it's been enabled everywhere and there haven't been any issues for a while we can then remove the old one completely. 
So TL;DR this choice of the default is to not change the behavior until the new chunk disk mapper has been proven.

Reply:
Regarding your suggestion to change that comment, that part is actually not really accurate `(asynchronous, used in upstream Prometheus)` because the new chunk disk mapper also works synchronously by default, only if the queue size gets set to a value above `0` then it switches to async mode.

How about this:

```
// Temporary flag which we use to select whether to use the new (used in upstream
// Prometheus) or the old (legacy) chunk disk mapper
```

Reply:
I updated the comment: 55cbbafe3

Reply:
Thanks.